service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/wrapper/ConditionWrapper.java

```diff
@@ -0,0 +1,121 @@
+package uk.nhs.adaptors.gp2gp.ehr.mapper.wrapper;
+
+import static uk.nhs.adaptors.gp2gp.ehr.mapper.wrapper.RelatedProblemWrapper.RELATED_PROBLEM_HEADER;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.hl7.fhir.dstu3.model.Annotation;
+import org.hl7.fhir.dstu3.model.Condition;
+import org.hl7.fhir.dstu3.model.Reference;
+
+import lombok.extern.slf4j.Slf4j;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.CodeableConceptCdMapper;
+import uk.nhs.adaptors.gp2gp.ehr.mapper.MessageContext;
+
+@Slf4j
+public class ConditionWrapper {
+
+    private static final String PROBLEM_INFO_PREFIX = "Problem Info: ";
+    private static final String PROBLEM_NOTES_PREFIX = "Problem Notes: ";
+    private static final String NOTES_DELIMITER = "; ";
+    private static Map<String, String> reversedRelMapping =
```

Comment:
Optional could use unmodifiable or immutable map here https://www.baeldung.com/java-immutable-maps

Reply:
Done. It would be even more helpful when that variable was exposed, however here we use only internally in a class. But applied the change as good practice.